nixos/modules/services/hardware/fancontrol.nix

```
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.fancontrol;
+  configFile = pkgs.writeText "fan.conf" cfg.config;
+
+in {
+
+  options.hardware.fancontrol = {
+    enable = mkEnableOption "fancontrol (requires fancontrol.config)";
+
+    config = mkOption {
+      type = types.lines;
+      default = /etc/fancontrol;
```

```suggestion
      default = null;
```
That way it creates an error when left empty.

---

woeps

---

why is null preferred? should i adjust to include `type.nullOr`?, then it fails quietly
with `default = "";`, the service fails and shows `please check your config file!`

---

As far as I understand it does not make sense to run it with an empty config file. In that case it is better if `nixos-rebuild` fails right away.